### PR TITLE
Solving #4

### PR DESCRIPTION
**Pull request by AntonBoch1244**
Why not to try solve JFF #4 issue? :smiley:

### CODE REVIEW

**Comment by AntonBoch1244** [comment]
Some tests are required. (Not on only my side.)

**Comment by w23** [comment]
I'm not sure I understand how it works. There should be a loop that restarts the whole thing after source is updated. I don't see one.

**Comment by AntonBoch1244** [comment]
@w23, I add documentation in Makefile, and for now restored some content to protoloop.sh file.

I search better solution for loop making by using Makefile.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+LDFLAGS :=-lGL -lX11 -lXfixes -lasound -pthread -lm
+INCLUDES:=-Iatto/include
+.SILENT : clean distclean atto/src/app_linux.c atto/src/app_x11.c
+.DEFAULT: run
+
+# run (or default) target calls proto target
+run: proto
+
+# this targets detects list of compilable files with includings and makes objects
+proto.c.o atto/src/app_linux.c.o atto/src/app_x11.c.o: proto.c atto/src/app_linux.c atto/src/app_x11.c
```
Comment (w23) [inline Makefile]: Build artefacts should be in a separate directory. See makefiles from atto, OpenSource, stameska, etc..

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	gcc -c ${INCLUDES} `echo $@ | sed 's/\.o//g'` -o $@
+
+# this targets calls git for update submodule atto
+atto/src/app_linux.c atto/src/app_x11.c:
+	git submodule update --recursive
```
Comment (w23) [inline Makefile]: Clever use of git submodules init, but I wonder whether that will try to update it on every build.

File: protoloop.sh [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
 while [ true ]
 do
-	cc $CFLAGS -lGL -lX11 -lXfixes -lasound -pthread -lm $SOURCES -o proto
-	if [ $? -eq 0 ]
+	make -k all
+	if [ "$?" -eq "0" ]
 	then
 		./proto &
-		PID=$!
+		echo $! > ./.pid
```
Comment (w23) [inline protoloop.sh:29]: Why replace PID envvar with a file? I find `$PID` more concise

Comment (AntonBoch1244) [reply protoloop.sh:29]: In future i wanna replace this shell script to just calling make. But for now i not have time to make better solution.
1. I solve `while`, as just call make rule in make rule. (but not applied patch for this)
A ⇒ 2. `.pid` I make as temp solution. Fully forget about `-e` in make.

**Pull request closed without merging**